### PR TITLE
feat: Add keyboard navigation to multiselect dropdown

### LINKED ISSUE #10373: [Task]: Make MultiSelectControl keyboard accessible
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

example: Check FilePicker widget's Allowed File type option in property control

Screenshot 2022-01-13 at 11 20 25 AM

### PR DESCRIPTION
**Pull request by aswathkk**
## Description

Use `Tab` to focus the dropdown
`Enter` or `Space` to open dropdown
`Arrow Up` and `Arrow Down` keys to focus dropdown menu items
`Space` or `Enter` to select items
`Esc` to close dropdown menu

Fixes #10373 

## Type of change

- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Manually tested and Jest tests added

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: feat/keyboard-accessible-multiselect-dropdown 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 55.23 **(0.12)** | 37.06 **(0.19)** | 35.29 **(0.3)** | 55.71 **(0.12)**
 :green_circle: | app/client/src/components/ads/MultiselectDropdown.tsx | 83.87 **(63.62)** | 70.83 **(53.33)** | 85.37 **(85.37)** | 82.14 **(60.52)**
 :red_circle: | app/client/src/utils/autocomplete/TernServer.ts | 52.71 **(-0.23)** | 40.83 **(-0.84)** | 36.21 **(0)** | 56.74 **(-0.25)**</details>

### CODE REVIEW

File: app/client/src/components/ads/MultiselectDropdown.tsx [hunk 1]
```diff
@@ -225,6 +232,52 @@ function MultiSelectDropdown(props: DropdownProps) {
...
+        if (isOpen) {
+          setCurrentItemIndex((prevIndex) => {
+            if (prevIndex <= 0) return props.options.length - 1;
+            return prevIndex - 1;
+          });
+        } else {
+          setIsOpen(true);
+        }
+        break;
+      case "ArrowDown":
```
Comment (SatishGandham) [inline app/client/src/components/ads/MultiselectDropdown.tsx:262]: I think these values are browser specific, please check and cover all possible cases
https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key

File: app/client/src/components/ads/MultiselectDropdown.tsx [hunk 2]
```diff
@@ -181,6 +186,8 @@ function MultiSelectDropdown(props: DropdownProps) {
     }
   }, []);
 
+  const [currentItemIndex, setCurrentItemIndex] = useState<number>(0);
+
   const optionClickHandler = (option: string) => {
```
Comment (SatishGandham) [inline app/client/src/components/ads/MultiselectDropdown.tsx:191]: Do we need useCallback here and at L235?

**Review by SatishGandham — CHANGES_REQUESTED** [review]

File: app/client/src/components/ads/MultiselectDropdown.tsx (hunk 1 above)
Comment (aswathkk) [reply app/client/src/components/ads/MultiselectDropdown.tsx:262]: Should we support browsers released before 2016?

**Review by SatishGandham — APPROVED** [review]

**Comment by vivekverma2312** [comment]
@aswathkk If we select any option using Enter/Space then we can not deselect it. If we deselect it then we can not select.
https://www.loom.com/share/3966f6f5fbbb4a9997395fd115decf5f

**Comment by vivekverma2312** [comment]
@aswathkk Now the focus of mouse click and Enter/Space is different. After I select any option using mouse the focus do not change.
https://www.loom.com/share/655fea8231e041b0a3319daae74b37ab

**Comment by vivekverma2312** [comment]
@aswathkk All issues are fixed and working fine now.

**Pull request merged**